src/prpy/planning/base.py

```diff
@@ -298,6 +318,23 @@ def plan(self, method, args, kw_args):
                     with Timer() as timer:
                         output = planner_method(*args, **kw_args)
 
+                    if not is_sequence_deterministic:
+                        # TODO: It is overly conservative to set _ENDPOINT,
+                        # e.g. for PlanToConfiguration. Unfortunately, there is
+                        # no easy way to detect this special case.
+                        SetTrajectoryTags(output, {
+                            Tags.DETERMINISTIC_TRAJECTORY: False,
+                            Tags.DETERMINISTIC_ENDPOINT: False,
+                        }, append=True)
+
+                        if not self._allow_nondeterministic:
+                            logger.warning(
+                                'Tagging trajectory as non-deterministic because an'
+                                ' earlier planner in the Sequence threw a'
+                                ' non-deterministic PlanningError. Pass the'
+                                ' "allow_nondeterministic" to this Sequence'
+                                ' constructor if you intended this behavior.')
+
```

Comment:
I'm a bit confused by `_allow_nondeterministic` flag. Why do we need this? You set the tags to `deterministic_traj = False, deterministic_endpoint = False` even if `self._allow_nondeterministic=False`.